lib/stateView/contract/SVContractMongoDbRepository.js

```diff
@@ -51,30 +95,64 @@ class SVContractMongoDbRepository {
    * Store SV Contract
    *
    * @param {SVContract} svContract
+   * @param {MongoDBTransaction} [transaction]
    * @returns {Promise}
    */
-  async store(svContract) {
+  async store(svContract, transaction = undefined) {
     const rawSVContract = svContract.toJSON();
-
     rawSVContract.contract = mongo.Binary(
       svContract.getContract().serialize(),
     );
 
-    return this.collection.updateOne(
-      { _id: svContract.getContractId() },
-      { $set: rawSVContract },
-      { upsert: true },
+    const filter = { _id: svContract.getContractId() };
+    const update = { $set: rawSVContract };
+    let updateOptions = { upsert: true };
```

Comment:
This shouldn't be an upsert, considering the fact that id is going to be an id of an identity and not the hash of the contract

Reply:
We were thinking to keep this part as is yet since we have validation logic in DPP.